terraform/aws/modules/storage/main.tf

```diff
@@ -107,15 +109,99 @@ resource "aws_s3_bucket" "km_blob_storage" {
   tags = merge(var.default_tags, {
     name = "km_blob_storage_${var.environment}"
   })
+
+  versioning {
+    enabled = true
+  }
+
+  versioning {
+    enabled = true
+  }
 }
```

Comment:
**Accurics remediation**
 
**[MEDIUM]: Ensure public access policies are not allowed for AWS S3 Buckets**
 
**Vulnerability**
AWS S3 buckets allow public access policies which may cause unauthorized access.
 
**Remediation**
Using Amazon S3 Block Public Access as a centralized way to limit public access. Block Public Access settings override bucket policies and object permissions. Be sure to enable Block Public Access for all accounts and buckets that you don't want publicly accessible.

source: accurics-